### PR TITLE
Added sys.os_name_human for Alpine, postmarketOS, OpenBSD and NetBSD

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: none
Changelog: title

### CODE REVIEW

**Comment by craigcomstock** [comment]
on PostmarketOS, /etc/os-release contains:

PRETTY_NAME="postmarketOS v23.12"
NAME="postmarketOS"
VERSION_ID="v23.12"
VERSION="v23.12"
ID="postmarketos"
ID_LIKE="alpine"
HOME_URL="https://www.postmarketos.org/"
SUPPORT_URL="https://gitlab.com/postmarketOS"
BUG_REPORT_URL="https://gitlab.com/postmarketOS/pmaports/issues"
LOGO="postmarketos-logo"

**Comment by craigcomstock** [comment]
As seen in /etc/os-release, postmarketOS is capitalized with a lowerase p at the beginning

**Review by larsewi — DISMISSED** [review]

**Comment by craigcomstock** [comment]
Oops, got the order wrong and missed the derived-from. While I'm at it, might as well add some more BSD that I can test.

File: libenv/sysinfo.c [hunk 1]
```diff
@@ -3474,14 +3474,38 @@ static void SysOSNameHuman(EvalContext *ctx)
...
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
                                       "FreeBSD", CF_DATA_TYPE_STRING,
-                                      "source=agent,derived-from=freebsd");
+                                      "source=agent");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "openbsd") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "OpenBSD", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived-from=netbsd");
```
Comment (olehermanse) [inline libenv/sysinfo.c:3483]:
```suggestion
                                      "source=agent,derived-from=openbsd");
```

File: libenv/sysinfo.c [hunk 2]
```diff
@@ -3474,14 +3474,38 @@ static void SysOSNameHuman(EvalContext *ctx)
...
+    }
+    else if (EvalContextClassGet(ctx, NULL, "netbsd") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "NetBSD", CF_DATA_TYPE_STRING,
+                                      "source=agent");
```
Comment (olehermanse) [inline libenv/sysinfo.c:3489]:
```suggestion
                                      "source=agent,derived-from=netbsd");
```

File: libenv/sysinfo.c [hunk 3]
```diff
@@ -3474,14 +3474,38 @@ static void SysOSNameHuman(EvalContext *ctx)
...
     {
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
                                       "Solaris", CF_DATA_TYPE_STRING,
                                       "source=agent,derived-from=solaris");
     }
+    else if (EvalContextClassGet(ctx, NULL, "postmarketos") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "postmarketOS", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived-from=alpine");
```
Comment (olehermanse) [inline libenv/sysinfo.c:3501]:
```suggestion
                                      "source=agent,derived-from=postmarketos");
```

File: libenv/sysinfo.c [hunk 4]
```diff
@@ -3474,14 +3474,38 @@ static void SysOSNameHuman(EvalContext *ctx)
...
+    }
+    else if (EvalContextClassGet(ctx, NULL, "alpine") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Alpine", CF_DATA_TYPE_STRING,
+                                      "source=agent");
```
Comment (olehermanse) [inline libenv/sysinfo.c:3507]:
```suggestion
                                      "source=agent,derived-from=alpine");
```

**Review by olehermanse — CHANGES_REQUESTED** [review]
`derived-from` is used to indicate which class caused you to set this value.

**Comment by craigcomstock** [comment]
> `derived-from` is used to indicate which class caused you to set this value.

makes sense. for some reason I saw it differently and was thinking about lineage and derivation like ubuntu derived-from debian ;) Thanks. :)

**Review by olehermanse — APPROVED** [review]

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10519/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10519/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10519/

**Review by larsewi — APPROVED** [review]

**Review by olehermanse — CHANGES_REQUESTED** [review]
Please update commit with:

```
Changelog: Title
```

Some open source users of these OSes might find this useful.

**Review by larsewi — APPROVED** [review]

**Review by olehermanse — APPROVED** [review]

**Pull request merged**